astropy/nddata/nddata.py

```diff
@@ -297,6 +309,16 @@ def wcs(self, wcs):
             raise TypeError("The wcs argument must implement either the high or"
                             " low level WCS API.")
 
+    @property
+    def psf(self):
+        return self._psf
+
+    @psf.setter
+    def psf(self, value):
+        if value is not None and not isinstance(value, np.ndarray):
```

Comment:
Does the requirement that PSF be an image have to made here or it could it be done in `CCDData`? I suppose the PSF could be the table or something, and it isn't until the PSF is written/read from `CCDData`  that it really has to be be an `ndarray`.

My thinking here is to keep the top level `NDData` as general as possible.

Reply:
I think that until we have a more generic way of representing PSFs, we should ensure that it is an array everywhere. Even the nddata docstring says `image representing the psf`, which I think is what we want? That said, I don't actually know how the nddata vs. ccddata functionality divide is actually intended.